java/src/main/java/ai/rapids/cudf/DType.java

```diff
@@ -17,76 +17,202 @@
 
 import java.util.EnumSet;
 
-public enum DType {
-  EMPTY(0, 0, "NOT SUPPORTED"),
-  INT8(1, 1, "byte"),
-  INT16(2, 2, "short"),
-  INT32(4, 3, "int"),
-  INT64(8, 4, "long"),
-  UINT8(1, 5, "uint8"),
-  UINT16(2, 6, "uint16"),
-  UINT32(4, 7, "uint32"),
-  UINT64(8, 8, "uint64"),
-  FLOAT32(4, 9, "float"),
-  FLOAT64(8, 10, "double"),
-  /**
-   * Byte wise true non-0/false 0.  In general true will be 1.
-   */
-  BOOL8(1, 11, "bool"),
-  /**
-   * Days since the UNIX epoch
-   */
-  TIMESTAMP_DAYS(4, 12, "date32"),
+public final class DType {
+
+  public enum DTypeEnum {
+    EMPTY(0, 0, "NOT SUPPORTED"),
+    INT8(1, 1, "byte"),
+    INT16(2, 2, "short"),
+    INT32(4, 3, "int"),
+    INT64(8, 4, "long"),
+    UINT8(1, 5, "uint8"),
+    UINT16(2, 6, "uint16"),
+    UINT32(4, 7, "uint32"),
+    UINT64(8, 8, "uint64"),
+    FLOAT32(4, 9, "float"),
+    FLOAT64(8, 10, "double"),
+    /**
+     * Byte wise true non-0/false 0.  In general true will be 1.
+     */
+    BOOL8(1, 11, "bool"),
+    /**
+     * Days since the UNIX epoch
+     */
+    TIMESTAMP_DAYS(4, 12, "date32"),
+    /**
+     * s since the UNIX epoch
+     */
+    TIMESTAMP_SECONDS(8, 13, "timestamp[s]"),
+    /**
+     * ms since the UNIX epoch
+     */
+    TIMESTAMP_MILLISECONDS(8, 14, "timestamp[ms]"),
+    /**
+     * microseconds since the UNIX epoch
+     */
+    TIMESTAMP_MICROSECONDS(8, 15, "timestamp[us]"),
+    /**
+     * ns since the UNIX epoch
+     */
+    TIMESTAMP_NANOSECONDS(8, 16, "timestamp[ns]"),
+
+    //We currently don't have mappings for duration type to I/O files, and these
+    //simpleNames might change in future when we do
+    DURATION_DAYS(4, 17, "int32"),
+    DURATION_SECONDS(8, 18, "int64"),
+    DURATION_MILLISECONDS(8, 19, "int64"),
+    DURATION_MICROSECONDS(8, 20, "int64"),
+    DURATION_NANOSECONDS(8, 21, "int64"),
+    //DICTIONARY32(4, 22, "NO IDEA"),
+
+    STRING(0, 23, "str"),
+    LIST(0, 24, "list"),
+    DECIMAL32(4, 25, "decimal32"),
+    DECIMAL64(8, 26, "decimal64"),
```

Comment:
decimal32 and decimal64 are not supported by CSV for parsing so we should not set a string indicating that they are. I have filed a follow on issue because this is a bug in other places in this code too.  We really should modify DType to just not have the strings here, and have the mapping closer to where it really matters.

https://github.com/rapidsai/cudf/issues/6574

Reply:
Hi @revans2 , is it mean that we need to modify method `csv_reader_options::builder.dtypes` to pass multiple arguments rather than a single string for data type representation?

Reply:
@revans2 please correct me if I am wrong but I think what you are saying is that we should leave the string blank or indicate in the string that its not supported yet. 

e.g. `DECIMAL32(4, 25, "NOT SUPPORTED")` 

Moving the string to a schema may be out of scope for this PR but that's a good point. Should we create a follow-on for that @revans2?